src/index.md

```diff
@@ -14,15 +14,40 @@ in the officially maintained [Rust Book](https://doc.rust-lang.org/book/).
 To read the book about gdnative (Godot 3 binding), follow [this link](../gdnative-book).
 
 
+## The basics
+
+A **library** is in its most basic form is a **collection of functions**. Godot has several of its own built-in libraries, but like most other programs made as a creation tool, it also has the capacity to access collections of libraries made outside of Godot. Those are called **extensions**, and like most other software, they usually come in separate files -- typically in the form of file extensions such as `.so` if you're on Linux, `.dll` if you're on Windows, or `.dylib` on MacOS.
```

Comment:
From Discord:
> I would probably try to be strict with terminology, otherwise we might confuse readers rather than clear up:
> - "Library" should be used in the C sense, as static or dynamic library. I don't think for the chapter it's relevant that Godot itself uses libraries (virtually any software does). So I'd focus on the dynamic library -- the GDExtension binding.
> - "Collections" almost always refers to containers and data structures. 
> - Godot's C API is called "GDExtension" (not "GDExtensions"), but different language binding libraries (godot-cpp, godot-rust, etc.) can be used as "extensions".

Also twice "is" in first sentence.

Last sentence, "[extensions] come in separate files -- typically in the form of file extensions" sounds weird. The "like most other software, they usually come in separate files" contains not much relevant information, I'd just say something like:

> Those are called **extensions** and are distributed as dynamic libraries, with file endings `.so` on Linux, `.dll` on Windows or `.dylib` on macOS.